### PR TITLE
Decompile hall of fame

### PR DESCRIPTION
**Pull request by DizzyEggg**

### CODE REVIEW

File: data/graphics.s [hunk 1]
```diff
@@ -156,7 +156,31 @@ gUnknown_08C2CEE0:: @ 8C2CEE0
 	.incbin "baserom.gba", 0xc2cee0, 0x240
 
 gUnknown_08C2D120:: @ 8C2D120
-	.incbin "baserom.gba", 0xc2d120, 0x600
+	.incbin "baserom.gba", 0xc2d120, 552
```
Comment (ProjectRevoTPP) [inline data/graphics.s]: why are these in decimal?

File: src/hall_of_fame.c [hunk 2]
```diff
@@ -1,4 +1,1678 @@
...
+// todo: dump actual graphics
+static const u32 sHallOfFame_Pal[] =
+{
+    0, 0x109F63B0, 0x5B6E, 0, 0, 0, 0, 0x7FFF2108
+};
+
+// todo: dump actual graphics
+static const u32 sHallOfFame_Gfx[] =
+{
+    0x3A010,
```
Comment (ProjectRevoTPP) [inline src/hall_of_fame.c]: ew

File: src/hall_of_fame.c [hunk 3]
```diff
@@ -1,4 +1,1678 @@
...
+    .paletteTag = 1001,
+    .oam = &sOamData_85E53FC,
+    .anims = sSpriteAnimTable_85E548C,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = sub_81751A4
+};
+
+// todo: dump actual graphics
+static const u32 sHallOfFame_Pal[] =
```
Comment (ProjectRevoTPP) [inline src/hall_of_fame.c]: ew 2 electric boogaloo

Comment (DizzyEggg) [reply src/hall_of_fame.c]: im the decomp guy, not the dump grahpics guy

File: data/graphics.s (hunk 1 above)
Comment (DizzyEggg) [reply data/graphics.s]: because, that's how ida shows size

File: include/malloc.h [hunk 4]
```diff
@@ -5,6 +5,12 @@
 #define calloc(ct, sz) AllocZeroed((ct) * (sz))
 #define free Free
 
+#define FREE_AND_SET_NULL(ptr)          \
```
Comment (Diegoisawesome) [inline include/malloc.h]: Is this macro necessary?

File: src/hall_of_fame.c [hunk 5]
```diff
@@ -1,4 +1,1678 @@
...
+#include "trig.h"
+#include "rng.h"
+#include "event_data.h"
 
+struct HallofFameMon
+{
+    u32 tid;
+    u32 personality;
+    u16 species : 9;
+    u16 lvl : 7;
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: Remove spaces around colons

File: src/hall_of_fame.c (hunk 3 above)
Comment (Diegoisawesome) [reply src/hall_of_fame.c]: When you decomp, please extract as well.

File: src/hall_of_fame.c (hunk 2 above)
Comment (Diegoisawesome) [reply src/hall_of_fame.c]: Please don't do this.

File: src/hall_of_fame.c [hunk 6]
```diff
@@ -1,4 +1,1678 @@
...
+            return FALSE;
+        }
+        break;
+    }
+    return TRUE;
+}
+
+#define tDontSaveData       data[0]
+#define tDisplayedMonId     data[1]
+#define tPokesNumber        data[2]
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: Pokes->Mon

File: src/hall_of_fame.c [hunk 7]
```diff
@@ -1,4 +1,1678 @@
...
+
+struct HofGfx
+{
+    u16 state;
+    u8 field_2[16];
+    u8 tilemap1[0x1000];
+    u8 tilemap2[0x1000];
+};
+
+static EWRAM_DATA struct HallofFameMons *sHofMonsPtr = NULL;
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: Mons->Mon

File: src/hall_of_fame.c [hunk 8]
```diff
@@ -1,4 +1,1678 @@
...
+    u32 tid;
+    u32 personality;
+    u16 species : 9;
+    u16 lvl : 7;
+    u8 nick[10];
+};
 
+struct HallofFameMons
+{
+    struct HallofFameMon mons[6];
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: mons->mon

File: src/hall_of_fame.c [hunk 9]
```diff
@@ -1,4 +1,1678 @@
...
+struct HallofFameMon
+{
+    u32 tid;
+    u32 personality;
+    u16 species : 9;
+    u16 lvl : 7;
+    u8 nick[10];
+};
 
+struct HallofFameMons
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: HallofFameMons->HallofFameTeam

File: src/hall_of_fame.c [hunk 10]
```diff
@@ -1,4 +1,1678 @@
...
+        SetGpuReg(REG_OFFSET_BLDCNT, 0);
+        SetGpuReg(REG_OFFSET_BLDALPHA, 0);
+        SetGpuReg(REG_OFFSET_BLDY, 0);
+        sub_8174FAC();
+        gMain.state++;
+        break;
+    case 3:
+        if (!sub_8175024())
+        {
+            struct HallofFameMons *fameMons = (struct HallofFameMons*)(gDecompressionBuffer);
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: fameMons->fameTeam

File: src/hall_of_fame.c [hunk 11]
```diff
@@ -1,4 +1,1678 @@
...
+#undef tDisplayedPoke
+#undef tPokesNumber
+#undef tFrameCount
+#undef tPlayerSpriteID
+#undef tMonSpriteId
+
+#define tCurrTeamNo     data[0]
+#define tCurrPageNo     data[1]
+#define tCurrPokeId     data[2]
+#define tPokesNo        data[4]
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: tCurrPokeId->tCurrMonId, tPokesNo->tMonNo

File: src/hall_of_fame.c [hunk 12]
```diff
@@ -1,4 +1,1678 @@
...
+static const u8 sUnused0[8] = {};
+
+static const struct CompressedSpritePalette sHallOfFame_ConfettiSpritePalette =
+{
+    gContestConfetti_Pal, 1001
+};
+
+static const u8 sUnused1[8] = {};
+
+static const s16 sHallOfFame_MonsFullTeamPositions[6][4] =
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: Mons->Mon

File: src/hall_of_fame.c [hunk 13]
```diff
@@ -1,4 +1,1678 @@
...
+    else if (gMain.newKeys & DPAD_DOWN && gTasks[taskId].tCurrPokeId < gTasks[taskId].tPokesNo - 1) // change mon +1
+    {
+        gTasks[taskId].tCurrPokeId++;
+        gTasks[taskId].func = Task_HofPC_PrintMonInfo;
+    }
+}
+
+static void Task_HofPC_HandlePaletteOnExit(u8 taskId)
+{
+    struct HallofFameMons* fameMons;
```
Comment (Diegoisawesome) [inline src/hall_of_fame.c]: fameMons->fameTeam

File: include/malloc.h (hunk 4 above)
Comment (DizzyEggg) [reply include/malloc.h]: Yes, it's used basically everywhere when memory is freed.

**Pull request merged**